config/initializers/carrierwave.rb

```diff
@@ -0,0 +1,22 @@
+CarrierWave.configure do |config|
+
+  if Rails.env.test?
+    config.enable_processing = false
+  end
+
+  if Rails.env.development? || Rails.env.test?
+    config.storage = :file
+  else
+    config.storage = :fog
+    config.fog_credentials = {
+      provider:              'AWS',
+      aws_access_key_id:     ENV["aws_access_key_id"],
+      aws_secret_access_key: ENV["aws_secret_access_key"],
+      region:                ENV["aws_region"],
+    }
+    config.fog_directory  = 'odinsy'
+    config.fog_public     = false
+    config.fog_attributes = { 'Cache-Control' => "max-age=#{365.day.to_i}" }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.